packages/@aws-cdk/aws-route53/lib/vpc-endpoint-service-domain-name.ts

```diff
@@ -0,0 +1,198 @@
+import * as crypto from 'crypto';
+import { IVpcEndpointService } from '@aws-cdk/aws-ec2';
+import { Fn } from '@aws-cdk/core';
+import { AwsCustomResource, AwsCustomResourcePolicy, PhysicalResourceId } from '@aws-cdk/custom-resources';
+import { Construct } from 'constructs';
+import { IPublicHostedZone, TxtRecord } from '../lib';
+
+// v2 - keep this import as a separate section to reduce merge conflict when forward merging with the v2 branch.
+// eslint-disable-next-line
+import { Construct as CoreConstruct } from '@aws-cdk/core';
+
+/**
+ * Properties to configure a VPC Endpoint Service domain name
+ */
+export interface VpcEndpointServiceDomainNameProps {
+
+  /**
+   * The VPC Endpoint Service to configure Private DNS for
+   */
+  readonly endpointService: IVpcEndpointService;
+
+  /**
+   * The domain name to use.
+   *
+   * This domain name must be verifiably owned by this account, or otherwise
+   * delegated to this account.
```

Comment:
same as a previous comment, but can we add a sentence on what verifiable means / what we consider to be verifiable?